PyInstaller/utils/hooks/__init__.py

```diff
@@ -1174,6 +1174,36 @@ def collect_entry_point(name: str) -> Tuple[list, list]:
     return datas, imports
 
 
+def get_hook_config(hook_api, key):
+    """
+    Get user settings for hooks.
+    Args:
+        key:
+            A key for the config.
+    Returns:
+        The value for the config. ``None`` if not set.
+
+    The ``get_hook_config`` function will lookup settings in the
+    ``Analysis.hooksconfig`` dict.
+
+    The hook settings can be added to ``.spec`` file in the form of:
+        a = Analysis(["my-app.py"],
+            ...
+            hooksconfig = {
+                "icons": ["Adwaita"],
+                "themes": ["Adwaita"],
+                "languages": ["en_GB", "zh_CN"],
+            },
+            ...
+        )
+
+    """
+    config = hook_api.analysis.hooksconfig
+    value = None
+    if config and key in config:
+        value = config[key]
+    return value
+
```

Comment:
I'd still like to see some name-spacing for keys so that options for different packages don't collide. Libraries like pygame and pyarcade will probably both want to use the name *icons* and *themes* could be applicable to any GUI or WebUI library that includes them. Either a domain prefix naming convention:
```python
"gi.icons": ["Adwaita"],
```
Or nested structures:
```python
hooksconfig = {
    "gi": {
        "icons": ["Adwaita"],
        "themes": ["Adwaita"],
        "languages": ["en_GB", "zh_CN"],
    },
},
```

I'm not fussy which.